src/billo.php

```diff
@@ -95,8 +95,11 @@ function main() {
     if ($message->hasFlag(Storage::FLAG_SEEN)) {
       // continue;
     } else {
-      messageAddFlag($GLOBALS['mailStorage'], $id, $message, Storage::FLAG_SEEN);
+      if ($listsConfig->isMember($list, $message->getHeader('from'))) {
```

Comment:
Look closer at the link I sent you https://github.com/dschmidt/billo/blob/master/src/billo.php#L43

You need to handle an addressList here, as there could be multiple from values.

Apart from that: there is no $list in this scope.
You could probably handle this better here: https://github.com/dschmidt/billo/blob/08f5dfc12d4d19ea54a46674824ccc8c5ebfe504/src/billo.php#L26

with code similar to this:

```php
     $members = $listsConfig->getMembers($list->getEmail());                                                                                     
                                                                                                                                                 
     // Check at least one of the senders ("from") is subscribed to the list                                                                     
     $subscribed = false;                                                                                                                        
     foreach($message->getHeader('from')->getAddressList() as $from) {                                                                           
       if (in_array($from->getEmail(), $members)) {                                                                                              
         $subscribed = true;                                                                                                                     
       }                                                                                                                                         
     }                                                                                                                                           
     if(!$subscribed) {                                                                                                                          
       continue;                                                                                                                                 
     } 
```
Please test, I haven't :)